storage/uri.go

```diff
@@ -202,3 +266,184 @@ func Exists(u fyne.URI) (bool, error) {
+//
+// NOTE: if the URIs do not share the same scheme, then this method will
+// instead use a reader and a writer. This may have performance implications if
+// either or both URIs is remote. If you wish to avoid this, explicitly compare
+// the URI schemes before calling Copy().
+//
+// Copy is backed by the repository system - this function may call into either
+// a generic implementation, or into a scheme-specific implementation depending
+// on which storage repositories have been registered.
+//
+// Since 2.0.0
+func Copy(source fyne.URI, destination fyne.URI) error {
+	return fmt.Errorf("TODO: implement this function")
+}
+
+// RenameImpl returns a method that given two URIs, 'src' and 'dest' both of
+// the same scheme this will rename src to dest. This means the resource
+// referenced by src will be copied into the resource referenced by dest, and
+// the resource referenced by src will no longer exist after the operation is
+// complete.
+//
+// This method may fail in several ways:
+//
+// * Different permissions or credentials are required to perform the
+//   rename operation.
+//
+// * This URI scheme could represent some resources that can be renamed,
+//   but either the source, destination, or both are not resources
+//   that support renaming.
+//
+// * Performing the rename operation depended on a lower level operation
+//   such as network or filesystem access that has failed in some way.
+//
+// Copy is backed by the repository system - this function may call into either
+// a generic implementation, or into a scheme-specific implementation depending
+// on which storage repositories have been registered.
+//
+// If both source and destination are not of the same URI scheme, then this
+// method will instead use a reader, write, and a call to Delete(), which may
+// have performance implications if either or both URIs are remote. If you wish
+// to avoid this, explicitly compare the scheme of both URIs before calling
+// Rename().
+//
+// Since 2.0.0
+func Rename(source fyne.URI, destination fyne.URI) error {
+	return fmt.Errorf("TODO: implement this function")
+}
+
+// Listable  will determine if the URI is listable or not.
+//
+// The returned method may fail in several ways:
+//
+// * Different permissions or credentials are required to check if the
+//   URI supports listing.
+//
+// * This URI scheme could represent some resources that can be listed,
+//   but this specific URI is not one of them (e.g. a file on a
+//   filesystem, as opposed to a directory).
+//
+// * Checking for listability depended on a lower level operation
+//   such as network or filesystem access that has failed in some way.
+//
+// Listable is backed by the repository system - this function either calls
+// into a scheme-specific implementation from a registered repository, or fails
+// with a URIOperationNotSupported error.
+//
+// Since 2.0.0
+func Listable(u fyne.URI) (bool, error) {
```

Comment:
`Listable` feels almost an assertion, would `CanList` be OK? It feels less ambiguous.

Reply:
Or `IsListable`

Reply:
That crossed my mind to, though generally we prefer to drop "Get" and "Is" prefixes.